tests/components/modbus/test_modbus.py

```diff
@@ -17,56 +31,512 @@
+                    CONF_TIMEOUT: 30,
+                    CONF_DELAY: 10,
+                }
+            ]
         },
     ],
 )
+async def test_config_modbus(hass, do_config):
+    """Run test for modbus."""
+
+    await _config_helper(hass, do_config)
+    assert DOMAIN in hass.data
+    assert len(hass.data[DOMAIN]) == 1
+
+
 @pytest.mark.parametrize(
     "do_config",
     [
         {
-            CONF_TYPE: "tcp",
-            CONF_HOST: "modbusTestHost",
-            CONF_PORT: 5501,
-        },
-        {
-            CONF_TYPE: "serial",
-            CONF_BAUDRATE: 9600,
-            CONF_BYTESIZE: 8,
-            CONF_METHOD: "rtu",
-            CONF_PORT: "usb01",
-            CONF_PARITY: "E",
-            CONF_STOPBITS: 1,
+            DOMAIN: [
+                {
+                    CONF_TYPE: "tcp",
+                    CONF_HOST: "modbusTestHost",
+                    CONF_PORT: 5501,
+                    CONF_NAME: "modbusTest1",
+                },
+                {
+                    CONF_TYPE: "tcp",
+                    CONF_HOST: "modbusTestHost",
+                    CONF_PORT: 5501,
+                    CONF_NAME: "modbusTest2",
+                },
+                {
+                    CONF_TYPE: "serial",
+                    CONF_BAUDRATE: 9600,
+                    CONF_BYTESIZE: 8,
+                    CONF_METHOD: "rtu",
+                    CONF_PORT: "usb01",
+                    CONF_PARITY: "E",
+                    CONF_STOPBITS: 1,
+                    CONF_NAME: "modbusTest3",
+                },
+            ]
         },
     ],
 )
-async def test_config_modbus(hass, do_discovery, do_options, do_config):
-    """Run test for modbus."""
+async def test_config_multiple_modbus(hass, do_config):
+    """Run test for multiple modbus."""
+
+    await _config_helper(hass, do_config)
+    assert DOMAIN in hass.data
+    assert len(hass.data[DOMAIN]) == len(do_config[DOMAIN])
+
+
+def _config_entry():
+    """Return class obj configured."""
+
+    config_hub = {
+        CONF_NAME: DEFAULT_HUB,
+        CONF_TYPE: "tcp",
+        CONF_HOST: "modbusTest",
+        CONF_PORT: 5001,
+        CONF_DELAY: 1,
+        CONF_TIMEOUT: 1,
+    }
+    hub = ModbusHub(config_hub)
+    assert hub.name == config_hub[CONF_NAME]
+    return hub
+
+
+async def test_class_ModbusHub_logger(hass, caplog):
+    """Run general test of class modbusHub."""
+
+    hub = _config_entry()
+    caplog.set_level(logging.DEBUG)
+    e = ModbusException("test logger")
+    caplog.clear()
+    hub._log_error(e)
+    assert len(caplog.records) == 1
+    assert caplog.records[0].levelname == "ERROR"
+
+    hub._log_error(e)
+    assert len(caplog.records) == 2
+    assert caplog.records[1].levelname == "DEBUG"
+
+
+async def test_pb_create_exception(hass):
+    """Run general test of class modbusHub."""
+
+    hub = _config_entry()
+    with mock.patch.object(modbus, "_LOGGER") as mock_logger, mock.patch(
```

Comment:
Use `caplog` ?

Reply:
ups! done

Reply:
It's not changed.

Reply:
Well I did change the one pointed out, now I changed all :-)